src/protocols/secure_channel/SimpleSessionResumptionStorage.cpp

```diff
@@ -0,0 +1,273 @@
+    uint16_t len = static_cast<uint16_t>(MaxStateSize());
+
+    DefaultStorageKeyAllocator keyAlloc;
+    ReturnErrorOnFailure(mStorage->SyncGetKeyValue(keyAlloc.SessionResumptionIndex(), buf, len));
+
+    TLV::ContiguousBufferTLVReader reader;
+    reader.Init(buf, len);
+
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Array, kIndexContentTag));
+    TLV::TLVType arrayType;
+    ReturnErrorOnFailure(reader.EnterContainer(arrayType));
+
+    size_t count = 0;
+    CHIP_ERROR err;
+    while ((err = reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag())) == CHIP_NO_ERROR)
+    {
+        if (count >= ArraySize(index.mNodes))
+        {
+            return CHIP_ERROR_NO_MEMORY;
+        }
+
+        TLV::TLVType containerType;
+        ReturnErrorOnFailure(reader.EnterContainer(containerType));
+
+        FabricIndex fabricIndex;
+        ReturnErrorOnFailure(reader.Next(kFabricIndexTag));
+        ReturnErrorOnFailure(reader.Get(fabricIndex));
+
+        NodeId peerNodeId;
+        ReturnErrorOnFailure(reader.Next(kPeerNodeIdTag));
+        ReturnErrorOnFailure(reader.Get(peerNodeId));
+
+        index.mNodes[count++] = ScopedNodeId(peerNodeId, fabricIndex);
+
+        ReturnErrorOnFailure(reader.ExitContainer(containerType));
+    }
+
+    if (err != CHIP_END_OF_TLV)
+    {
+        return err;
+    }
+
+    ReturnErrorOnFailure(reader.ExitContainer(arrayType));
+    ReturnErrorOnFailure(reader.VerifyEndOfContainer());
+
+    index.mSize = count;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SimpleSessionResumptionStorage::SaveLink(ConstResumptionIdView resumptionId, const ScopedNodeId & node)
+{
+    // Save a link from resumptionId to node, in key: /f/<fabricIndex>/r/<resumptionId>
+    uint8_t buf[MaxScopedNodeIdSize()];
+
+    TLV::TLVWriter writer;
+    writer.Init(buf);
+
+    TLV::TLVType outerType;
+    ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, outerType));
+    ReturnErrorOnFailure(writer.Put(kFabricIndexTag, node.GetFabricIndex()));
+    ReturnErrorOnFailure(writer.Put(kPeerNodeIdTag, node.GetNodeId()));
+    ReturnErrorOnFailure(writer.EndContainer(outerType));
+
+    const auto len = writer.GetLengthWritten();
+    VerifyOrDie(CanCastTo<uint16_t>(len));
+
+    DefaultStorageKeyAllocator keyAlloc;
+    ReturnErrorOnFailure(mStorage->SyncSetKeyValue(StorageKey(keyAlloc, resumptionId), buf, static_cast<uint16_t>(len)));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SimpleSessionResumptionStorage::LoadLink(ConstResumptionIdView resumptionId, ScopedNodeId & node)
+{
+    uint8_t buf[MaxScopedNodeIdSize()];
+    uint16_t len = static_cast<uint16_t>(MaxStateSize());
+
+    DefaultStorageKeyAllocator keyAlloc;
+    ReturnErrorOnFailure(mStorage->SyncGetKeyValue(StorageKey(keyAlloc, resumptionId), buf, len));
+
+    TLV::ContiguousBufferTLVReader reader;
+    reader.Init(buf, len);
+
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
```

Comment:
We already had these bits in Save/LoadIndex.  Should have a helper method to write/read a ScopedNodeId here.